packages/react/menu/src/Menu.tsx

```diff
@@ -710,8 +743,31 @@ const MenuSubTrigger = React.forwardRef((props, forwardedRef) => {
         data-state={getOpenState(context.open)}
         {...props}
         ref={composeRefs(forwardedRef, context.onTriggerChange)}
-        onMouseMove={composeEventHandlers(props.onMouseMove, () => {
-          if (!props.disabled) context.onOpenChange(true);
+        onMouseMove={composeEventHandlers(props.onMouseMove, (event) => {
+          if (contentContext.isMouseMovingToSubmenu(event)) return;
+          if (!props.disabled && !context.open) {
+            openTimersRef.current.push(window.setTimeout(() => context.onOpenChange(true), 100));
+          }
+        })}
+        onMouseLeave={composeEventHandlers(props.onMouseLeave, (event) => {
+          openTimersRef.current.map(window.clearTimeout);
+          openTimersRef.current = [];
+
+          const contentRect = context.content?.getBoundingClientRect();
+          if (contentRect) {
+            contentContext.onPointerGraceAreaChange([
+              { x: event.clientX, y: event.clientY },
+              { x: contentRect.left, y: contentRect.top },
+              { x: contentRect.left, y: contentRect.bottom },
```

Comment:
Are we sure we don't need to flip `x` when RTL? I know we tested earlier today but having another look things don't feel quite right in that configuration.

Reply:
Ah good spot, yeah, looks like we would need `contentRect.right` here somewhere in rtl

Reply:
I think you're right, I think they should x should be `contentRect.right`, oh and actually now that I think about it, it's more complicated than that, it depends on the actual side it gets set on (collisions)

Reply:
Bit of a hack for now because it's deep down, but we should be able to use the `content` and read the `data-side` on it to get the final placed side.

Reply:
Not too bothered about the hack because all the positioning logic will get swapped soon.

Reply:
I'd be ok with that considering it's changing soon, my biggest worry though is remembering to come back to make the change. How do we usually track short term fix regressions? I guess more tests? 😅

Reply:
I'll remember 😀